### ISSUE TITLE
Service entry https/tls + destination rule simple TLS - interactions

### ISSUE THREAD

**Issue by rshriram** [istio/istio#17031]
When a service entry port has HTTPS,  then we are setting up a listener with sni match. In this scenario, from a usability perspective, when the user specifies a destination rule for that service entry with TLS mode simple, we should consider ignoring that tls mode because the inbound traffic is already using HTTPS - we simply need to forward it.

e.g.,

```yaml
ServiceEntry:
host: google.com
ports:
- 443, https
- 80, http
---
DestinationRule:
host: google.com
trafficPolicy:
  tls:
    mode: SIMPLE
    # nothing else is specified
```

Today, we would be generating a cluster for google.com for port 443 with upstream TLS context. 
the question is, should we skip generating the upstream TLS context, so that we can eliminate some user pain?

**Comment by rshriram** [comment]
cc @louiscryan / @howardjohn / @hzxuzhonghu thoughts?

**Comment by hzxuzhonghu** [comment]
make sense

**Comment by hzxuzhonghu** [comment]
I applied the below config
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: google
spec:
  hosts:
  - "www.google.com"
  location: MESH_EXTERNAL
  ports:
  - number: 443
    name: https
    protocol: HTTPS
  resolution: NONE
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: google
spec:
  host: www.google.com
  trafficPolicy:
    tls:
      mode: SIMPLE

```
And got an error when curl www.google.com

```
$ curl https://www.google.com

curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
```

**Comment by hzxuzhonghu** [comment]
With #17107, no error met

**Comment by hzxuzhonghu** [comment]
It does not work for the below config with error
```
curl http://www.google.com 
upstream connect error or disconnect/reset before headers. reset reason: connection termination
```

```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: google
spec:
  hosts:
  - www.google.com
  ports:
  - number: 80
    name: http
    protocol: HTTP
  - number: 443
    name: https
    protocol: HTTPS
  resolution: DNS
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: rewrite-port-for-google
spec:
  hosts:
  - www.google.com
  http:
  - match:
    - port: 80
    route:
    - destination:
        host: www.google.com
        port:
          number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: originate-tls-for-google
spec:
  host: www.google.com
  trafficPolicy:
    loadBalancer:
      simple: ROUND_ROBIN
    portLevelSettings:
    - port:
        number: 443
      tls:
        mode: SIMPLE # initiates HTTPS when accessing www.google.com
```

**Comment by hzxuzhonghu** [comment]
It seems we can not achieve both http and https with a single cluster

**Comment by vadimeisenbergibm** [comment]
@rshriram Please note that the port 443 does not necessarily means that the traffic arrives to the port 443 with TLS already originated. Consider the example at https://github.com/istio/istio/issues/17031#issuecomment-531760062. The user issues an HTTP request to the port 80, Envoy redirects it to port 443 and performs TLS origination. This is how TLS origination is performed in Istio today.

**Comment by howardjohn** [comment]
@vadimeisenbergibm I just sent a PR out s few days ago that blocks all http traffic in port 443, since doing so will break outbound traffic. Are you saying that will break this use case?

**Comment by vadimeisenbergibm** [comment]
@howardjohn yes, it will break this use case. We should probably start using some other port at the egress gateway for TLS origination. We can allocate the ports for the egress gateway in the following way:

Port 80: HTTP in, HTTP out
Port 443: TLS in, TLS out
Port 8443 (or some other port): HTTP in, TLS out

And then we should update the Istio deployment to add the port for tls origination to the istio-egressgateway service and update the docs.

**Comment by howardjohn** [comment]
Ok so that's purely a docs change? We just need to do something very soon since my change landed in master and 1.4 is cut  next week
